Utilities/File.cpp

```diff
@@ -1207,6 +1209,78 @@ const std::string& fs::get_executable_dir()
 	return s_dir;
 }
 
+std::string fs::get_data_dir(const std::string& prefix, const std::string& location, const std::string& suffix)
+{
+	static const std::string s_dir = []
+	{
+		const std::string& dir = get_config_dir() + "/data/";
+
+		if (!is_dir(dir) && !create_path(dir))
+		{
+			return get_config_dir();
+		}
+
+		return dir;
+	}();
+
+	std::vector<u8> buf;
+	buf.reserve(location.size() + 1);
+
+	// Normalize location
+	for (char c : location)
+	{
+#ifdef _WIN32
+		if (c == '/' || c == '\\')
+#else
+		if (c == '/')
+#endif
+		{
+			if (buf.empty() || buf.back() != '/')
+			{
+				buf.push_back('/');
+			}
+
+			continue;
+		}
+		
+		buf.push_back(c);
+	}
+
+	// Calculate hash
+	u8 hash[20];
+	sha1(buf.data(), buf.size(), hash);
+
+	// Concatenate
+	std::string&& result = fmt::format("%s%s/%016llx%08x-%s/", s_dir, prefix, reinterpret_cast<be_t<u64>&>(hash[0]), reinterpret_cast<be_t<u32>&>(hash[8]), suffix);
+
+	if (!is_dir(result))
+	{
+		// Create dir if necessary
+		if (create_path(result))
+		{
+			// Acknowledge original location
+			file(result + ".location", rewrite).write(buf);
+		}
+	}
+
+	return result;
+}
+
+std::string fs::get_data_dir(const std::string& prefix, const std::string& path)
+{
+#ifdef _WIN32
+	const auto& delim = "/\\";
+#else
+	const auto& delim = "/";
```

Comment:
this should work in windows too provided you are not using \??\C:\whatever extended paths